src/vmm/src/arch/aarch64/vcpu.rs

```diff
@@ -106,6 +107,22 @@ pub fn setup_boot_regs(
         vcpufd
             .set_one_reg(id, &get_fdt_addr(mem).to_le_bytes())
             .map_err(|err| VcpuError::SetOneReg(id, err))?;
+
+        // Reset the physical counter for the guest. This way we avoid guest reading
+        // host physical counter.
+        // Resetting KVM_REG_ARM_PTIMER_CNT for singe vcpu is enough because there is only
+        // one timer struct with offsets per VM.
+        // Because the access to KVM_REG_ARM_PTIMER_CNT is only present starting 6.4 kernel,
+        // we don't fail if ioctl returns an error.
+        // Note: the value observed by the guest will still be above 0, because there is a delta
+        // time between this resetting and first call to KVM_RUN
+        let zero: u64 = 0;
+        if vcpufd
+            .set_one_reg(KVM_REG_ARM_PTIMER_CNT, &zero.to_le_bytes())
+            .is_err()
+        {
+            warn!("Unable to reset performance counter. VM will use host value instead.");
```

Comment:
I really don't think we should have warnings that the user cannot silence. Maybe we can only print this if running on 6.4 or newer? Rest lgtm

Reply:
I don't think this is needed to be silenced. 6.4 is not super new and as time goes on, the less users will use kernel which will print the warning. If we add option to silence this, we will have to have it until we deprecate it and this is more work to do.

Reply:
Well, this is a warning and not an error. They are not meant to be "fixed" I think.
I can add a link to the patch series that added the needed fixes to the changelog and tell users to use kernel with this patches if they want to remove the warning.

Reply:
feels like we could update https://github.com/firecracker-microvm/firecracker/blob/main/docs/prod-host-setup.md#arm-only-physical-counter-directly-passed-through-to-the-guest in order to explain that

Reply:
I was also thinking about making it `info`, but then only people who enable `info` log level will see it.

Reply:
how about we only warn if the `KVM_CAP_COUNTER_OFFSET` KVM capability is supported (meaning we are on a kernel where setting the registers should be possible)?

Reply:
I think if it's supported then it should error as any normal register setting. So summarizing how I think it should work:

```
set_reg(KVM_REG_ARM_PTIMER_CNT, 0)
if is_err():
    if supported(KVM_CAP_COUNTER_OFFSET) :
        log.error(...)
    else
        log.info|warn(cannot set counter because capability is missing, <link to FC docs>)
```

I am fine with either warn or info.

Reply:
So, do you guys think it is worth messing with capability checks? Or can we leave a warning as it is?

Reply:
Added couple of commits to split KVM from VM struct and do a proper optional capability checks.